File: 3859-maximum-product-of-two-digits/3859-maximum-product-of-two-digits.c
int maxProduct(int num) {
    int n = 0, m = 0;
    for (; num; ) {
        int r = num % 10;
        if (r >= n) {
            m = n;
            n = r;
        }
        else if (r > m)
            m = r;
        num /= 10;
    }
    return n * m;
}

File: 3859-maximum-product-of-two-digits/3859-maximum-product-of-two-digits.go
func maxProduct(num int) int {
    n, m := 0, 0
    for num > 0 {
        r := num % 10
        num /= 10
        
        if r >= n {
            m = n
            n = r
        } else if r > m {
            m = r
        }
    }
    return n * m
}